### DISCUSSION TITLE
Infinity reading

### DISCUSSION THREAD

**Discussion by Anthonybelui** [encode/httpcore#859] [Potential Issue]
Hello, thanks for developing the library.
I wanted to ask you to fix/define the logic for the read timeout in socks_proxy. This creates an infinity reading. 

File httpcore/_sync/socks_proxy.py
Function _init_socks5_connection()
Line: `incoming_bytes = stream.read(max_bytes=4096)` > `incoming_bytes = stream.read(max_bytes=4096, timeout=SomeTimeout)`

**Comment by lovelydinosaur** [comment]
Well spotted yea.
You could escalate this into an issue if you like... "Socks proxy should support read/write timeouts".

A pull request for this would be ace.
